File: plugin/src/test/java/testcode/ObjectDeserialization.java
package testcode;

import testcode.sqli.UserEntity;

import java.io.*;

public class ObjectDeserialization {


    public UserEntity deserializeObject(InputStream receivedFile) throws IOException, ClassNotFoundException {
        ObjectInputStream in = new ObjectInputStream(receivedFile);
        try {
            return (UserEntity) in.readObject();
        }
        finally {
            in.close();
        }
    }
}

File: findsecbugs-test-kotlin/src/test/kotlin/com/h3xstream/findsecbugs/deserialisation/ObjectDeserialization.kt
package com.h3xstream.findsecbugs.deserialisation

import org.apache.commons.io.input.ClassLoaderObjectInputStream

import java.io.*

class ObjectDeserialization {

    @Throws(IOException::class, ClassNotFoundException::class)
    fun deserializeObject(receivedFile: InputStream): UserEntity {
        val `in` = ObjectInputStream(receivedFile)
        try {
            return `in`.readObject() as UserEntity
        } finally {
            `in`.close()
        }
    }

    @Throws(IOException::class, ClassNotFoundException::class)
    fun deserializeObjectWithInheritance(receivedFile: InputStream): UserEntity {
        val `in` = ClassLoaderObjectInputStream(javaClass.classLoader, receivedFile)
        try {
            return `in`.readObject() as UserEntity
        } finally {
            `in`.close()
        }
    }
}